controller/orphan_controller.go

```diff
@@ -261,15 +261,27 @@ func (oc *OrphanController) reconcile(orphanName string) (err error) {
 		log.Infof("Orphan got new owner %v", oc.controllerID)
 	}
 
+	existingOrphan := orphan.DeepCopy()
+
 	if !orphan.DeletionTimestamp.IsZero() {
+		defer func() {
+			if reflect.DeepEqual(existingOrphan.Status, orphan.Status) {
+				return
+			}
+
+			if _, updateStatusErr := oc.ds.UpdateOrphanStatus(orphan); updateStatusErr != nil {
```

Comment:
What happens if the update is conflicted?

Reply:
The status update will be dropped. The cleanup errors will be logged but won't be not recorded in the conditions, and there is no other status change during the finalization. WDYT @derekbit ?

Reply:
After rethinking the logic, we can return `err` and retry. If it always fails to update the status, it means the API server is problematic. WDYT @c3y1huang @COLDTURNIP

Reply:
Sounds good to me.

Reply:
Agree.